services/venice-controller/src/main/java/com/linkedin/venice/controller/HelixVeniceClusterResources.java

```diff
@@ -206,6 +208,23 @@ public HelixVeniceClusterResources(
     veniceAdminStats = new VeniceAdminStats(metricsRepository, "venice-admin-" + clusterName);
     this.storagePersonaRepository =
         new StoragePersonaRepository(clusterName, this.storeMetadataRepository, adapterSerializer, zkClient);
+    if (!config.isParent() && config.isSystemStoreHealthCheckEnabled()) {
+      if (!admin.getPushStatusStoreReader().isPresent()) {
+        throw new VeniceException("Da Vinci push status reader is not enabled.");
+      }
+      if (!admin.getPushStatusStoreWriter().isPresent()) {
+        throw new VeniceException("Da Vinci push status writer is not enabled.");
+      }
+      this.systemStoreHealthCheckService = new SystemStoreHealthCheckService(
+          storeMetadataRepository,
+          admin.getMetaStoreReader(),
+          admin.getMetaStoreWriter(),
+          admin.getPushStatusStoreReader().get(),
+          admin.getPushStatusStoreWriter().get(),
+          config.getDaVinciPushStatusScanIntervalInSeconds());
+    } else {
+      this.systemStoreHealthCheckService = null;
```

Comment:
Yeah that's true.

Reply:
Actually no, the variable is final, so in each branch we need to set a value for once.

Reply:
Sure, keep it if you prefer, just a nit point as mentioned.

Reply:
Hmm, just want to clarify, this should not be a nit, it is required by Java compiler that a final variable is assigned on every path inside the constructor. Since the variable does not need to be re-assigned, I use final modifier here so it requires declaring the value on the "else" branch...

Reply:
Sure, understood. Alternative is to remove `final` and the `else` branch. But, declaring it as `final` also have benefit as well, so it's trade-off  :-)